addons/medical_engine/functions/fnc_getItemArmour.sqf

```diff
@@ -0,0 +1,58 @@
+#include "script_component.hpp"
+/*
+ * Author: Pterolatypus
+ * Scans config or uses cached values to determine what armour a particular item provides to a particular hitpoint
+ *
+ * Arguments:
+ * 0: Item Classname <STRING>
+ * 1: Hitpoint <STRING>
+ * 2: Disable Caching <BOOLEAN> (Default: false)
+ *
+ * Return Value:
+ * Protection the item applies to that hitpoint <ARRAY>
+ * 0: Armour <NUMBER>
+ * 1: ExplosionShielding <NUMBER>
+ *
+ * Example:
+ * ["V_PlateCarrier_rgr", "HitChest"] call ace_medical_engine_fnc_getItemArmour
+ *
+ * Public: No
+ */
+
+ params ["_item","_hitpoint",["_noCache",false]];
+
+_hitpoint = toLower _hitpoint;
+
+// Return the cached value if found
+private _cached = if (_noCache) then {[]} else {GVAR(armourCache_items) getVariable [_item,[]]};
+if !(_cached isEqualTo []) exitwith {
+    private _index = _cached findIf {_x#0 isEqualTo _hitpoint};
+    if (_index == -1) then {[0,0]} else {(_cached select _index)#1};
+};
+TRACE_1("Item armour cache miss",_item);
+
+private _protection = configFile >> "CfgWeapons" >> _item >> "ItemInfo" >> "HitpointsProtectionInfo";
+// If the item has no protection, cache as empty
+if (!isClass _protection) exitwith {
+    GVAR(armourCache_items) setVariable [_item,[]];
```

Comment:
Any benefit in setting this as empty array? Isn't this the default already? Could maybe save some RAM skiping this.

Reply:
I need to distinguish between "item not in cache -> scan config" and "item in cache but no values -> return default"